### PR TITLE
Fix clashing subuid

### PR DESCRIPTION
**Pull request by ashley-cui**
When initing a podman machine, we match core's UID to the UID of the user on the host. If the UID is in the subUID range, the machine throws an error. Check if the UID is within the default range for /etc/subuid (10000:1000000), and if it is, we adjust the range to not include it.

[NO NEW TESTS NEEDED] 
Mostly because this requires changing UID of the host and I don't think we should mess with it, but let me know if I should write a test.

Fixes: https://github.com/containers/podman/issues/17893

#### Does this PR introduce a user-facing change?

```release-note
Fixed a bug where the `podman machine init` command would create invalid machines when run with certain UID's
```

### CODE REVIEW

File: pkg/machine/ignition.go [hunk 1]
```diff
@@ -307,7 +307,12 @@ machine_enabled=true
 	delegateConf := `[Service]
 Delegate=memory pids cpu io
 `
-	subUID := `%s:100000:1000000`
+	// Prevent subUID from clashing with actual UID
+	subUID := 100000
+	if uid >= 100000 && uid <= 1100000 {
```
Comment (eriksjolund) [inline pkg/machine/ignition.go:312]: 1100000 is outside the range

```
%s:100000:1000000
```

```suggestion
	if uid >= 100000 && uid < 1100000 {
```

File: pkg/machine/ignition.go [hunk 2]
```diff
@@ -307,7 +307,12 @@ machine_enabled=true
 	delegateConf := `[Service]
 Delegate=memory pids cpu io
 `
-	subUID := `%s:100000:1000000`
+	// Prevent subUID from clashing with actual UID
+	subUID := 100000
```
Comment (rhatdan) [inline pkg/machine/ignition.go:311]: You could also boost this number to a far larger number, Say 2,000,000,000 And then it is less likely to conflict.

File: pkg/machine/ignition.go [hunk 3]
```diff
@@ -307,7 +307,12 @@ machine_enabled=true
...
+	if uid >= 100000 && uid < 1100000 {
```
Comment (rhatdan) [inline pkg/machine/ignition.go:312]: Suggest:
        if uid >= subUID && uid < (subUID+100000) {

You might want to make the 100000 become a constant as well
subUIDs=100000
        if uid >= subUID && uid < (subUID+subUIDs) {

File: pkg/machine/ignition.go (hunk 2 above)
Comment (ashley-cui) [reply pkg/machine/ignition.go:311]: I guess it's better to make sure it doesn't conflict for certain, so I think this logic is necessary regardless

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Comment by umohnani8** [comment]
LGTM

**Pull request merged**